File: src/lib/classes/TrackLink.ts
import { ExternalUrl } from "./ExternalUrl";

export class TrackLink {
  external_urls!: ExternalUrl;
  href!: string;
  id!: string;
  type!: 'track';
  uri!: string;

  private constructor() {
  }

  public static fromTrackLinkObject(tl: SpotifyApi.TrackLinkObject): TrackLink {
    const trackLink = new TrackLink();
    trackLink.external_urls = ExternalUrl.fromExternalUrlObject(tl.external_urls);
    trackLink.href = tl.href;
    trackLink.id = tl.id;
    trackLink.type = tl.type;
    trackLink.uri = tl.uri;
    return trackLink;
  }
}

File: src/main/java/de/wittenbude/expandify/models/pojos/TrackLink.java
package de.wittenbude.expandify.models.pojos;

import lombok.Data;
import lombok.NoArgsConstructor;
import se.michaelthelin.spotify.enums.ModelObjectType;
import se.michaelthelin.spotify.model_objects.specification.ExternalUrl;

@Data
@NoArgsConstructor
public class TrackLink {
    private ExternalUrl externalUrls;
    private String href;
    private String id;
    private ModelObjectType type;
    private String uri;

    public TrackLink(se.michaelthelin.spotify.model_objects.specification.TrackLink trackLink) {
        this.externalUrls = trackLink.getExternalUrls();
        this.href = trackLink.getHref();
        this.id = trackLink.getId();
        this.type = trackLink.getType();
        this.uri = trackLink.getUri();
    }
}
